tests/benchq/resource_estimation/test_openfermion_re.py

```diff
@@ -123,14 +135,160 @@ def test_monotonicity_of_duration_wrt_SCC_time(SCC_time_low, SCC_time_high):
 def test_linearity_of_duration_wrt_SCC_time(SCC_time_low, SCC_time_high):
     n_toffoli = 100
     n_logical_qubits = 100
+    physical_error_rate = 1.0e-4
     resource_estimates_low = get_physical_cost(
-        n_toffoli, n_logical_qubits, SCC_time_low
+        num_toffoli=n_toffoli,
+        num_logical_qubits=n_logical_qubits,
+        surface_code_cycle_time=SCC_time_low,
+        physical_error_rate=physical_error_rate,
     )
     resource_estimates_high = get_physical_cost(
-        n_toffoli, n_logical_qubits, SCC_time_high
+        num_toffoli=n_toffoli,
+        num_logical_qubits=n_logical_qubits,
+        surface_code_cycle_time=SCC_time_high,
+        physical_error_rate=physical_error_rate,
     )
 
     numpy.testing.assert_allclose(
-        resource_estimates_high.total_time_in_seconds / SCC_time_high.microseconds,
-        resource_estimates_low.total_time_in_seconds / SCC_time_low.microseconds,
+        resource_estimates_high.total_time_in_seconds / SCC_time_high.total_seconds(),
+        resource_estimates_low.total_time_in_seconds / SCC_time_low.total_seconds(),
+    )
+
+
+@pytest.mark.parametrize(
+    "num_toffoli,num_T",
+    [
+        (20, 20),
+        (40, 40),
+        (20, 30),
+    ],
+)
+def test_ratio_of_failure_prob(num_toffoli, num_T):
+    num_logical_qubits = 12
+    SCC_time = datetime.timedelta(microseconds=2)
+    physical_error_rate = 1.0e-4
+    best_toffoli = get_physical_cost(
+        num_logical_qubits=num_logical_qubits,
+        num_toffoli=num_toffoli,
+        surface_code_cycle_time=SCC_time,
+        physical_error_rate=physical_error_rate,
+    )
+    best_T = get_physical_cost(
+        num_logical_qubits=num_logical_qubits,
+        num_T=num_T,
+        surface_code_cycle_time=SCC_time,
+        physical_error_rate=physical_error_rate,
     )
+    """
+    This ascertains if ratio of failure rate of Toffoli
+    and T Magic State factories is 2:1
+    """
+    assert best_toffoli.extra.failure_prob / best_T.extra.failure_prob == 2.0
+
+
+@pytest.mark.parametrize(
+    "n_toffoli,n_T",
+    [
+        (20, 20),
+        (40, 40),
+    ],
+)
+def test_calc_of_Algorithm_Failure_Prob(n_toffoli, n_T):
```

Comment:
change `test_calc_of_Algorithm_Failure_Prob` to `test_calc_of_algorithm_failure_prob`

Reply:
Done